src/schema/provenance_schema.yaml

```diff
@@ -105,15 +105,31 @@ doi_properties: &doi_properties
     # So we can't define it as `immutable: true`
     type: string
     description: "The doi of the registered entity. e.g. 10.35079/hbm289.pcbm.487"
+    before_property_create_validators:
+     - verify_DOI_pair
+     - halt_update_if_DOI_exists
+     - halt_if_DOI_for_unpublished_dataset
```

Comment:
We should only run this `halt_if_DOI_for_unpublished_dataset` trigger for Collection, and skip it for Dataset.

Reply:
Modified `provenance_schema.yaml` and `schema_validators.py` to support validators             `halt_DOI_if_collection_missing_elements` and `halt_if_DOI_for_unpublished_dataset`.

Introduced `registered_doi` and `doi_url` under `ENTITIES: Collection: properties:`, as well as leaving under `doi_properties`, to enable these Collection-specific validations.